bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java

```diff
@@ -576,18 +584,13 @@ protected boolean thingIsInitialized() {
     }
 
     @Override
-    public void bridgeHandlerInitialized(ThingHandler thingHandler, Bridge bridge) {
-        // do nothing by default, can be overridden by subclasses
-    }
-
-    @Override
-    public void bridgeHandlerDisposed(ThingHandler thingHandler, Bridge bridge) {
-        // do nothing by default, can be overridden by subclasses
-    }
-
-    @Override
     public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
-        // do nothing by default, can be overridden by subclasses
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE
+                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE);
```

Comment:
Doing the refactoring of the OH2 bindings, this default implementation does not appear to be helpful.
In most cases, the Thing is simply expected to go ONLINE once the bridge is ONLINE. This means that this method needs to be overwritten, so the question arises if super. bridgeStatusChanged() should be called at all or if the logic isn't reimplemented. Both variants are not really nice.

I think we should change this line to ThingStatus.ONLINE as this will be the expected behavior in most cases. Whoever needs something else, should override the default impl. 
@sbussweiler WDYT?